pkg/schema/schema.go

```diff
@@ -153,12 +153,27 @@ func getValueFromAnn(defaultAnn *DefaultAnnotation, t Type) (interface{}, error)
 	defaultValue := defaultAnn.Val()
 	if node, ok := defaultValue.(yamlmeta.Node); ok {
 		defaultValue = node.DeepCopyAsInterface()
-		typeCheck = t.AssignTypeTo(defaultValue.(yamlmeta.Node))
+		val := defaultValue.(yamlmeta.Node)
+		typeCheck = t.AssignTypeTo(val)
+		if !typeCheck.HasViolations() {
+			typeCheck = CheckNode(val)
+		}
```

Comment:
Does it make sense to use `node.DeepCopy()`, here? That method returns a `yamlmeta.Node` already... no need to cast!

Reply:
When changing the code to the following:
```
val := node.DeepCopyAsNode()
typeCheck = t.AssignTypeTo(val)
```
The following tests fail:
TestSchemaInspect_exports_an_OpenAPI_doc/including_explicitly_set_default_values
TestSchemaInspect_exports_an_OpenAPI_doc/including_nullable_values_with_defaults

I have not investigated this further, but I would be interested if @pivotaljohn or @cari-lynn had any idea why this is happening?

Seems to not be adding the default `ree` key when the default value came from `@schema/default` ann.

Reply:
The tests are failing because we need to assign the copy _back_ to `defaultValue` (it's what's getting merged in).

I now realize that it's not as simple as just returning a `Node` because the static type of `defaultValue` is `interface{}`.

One might argue that the following is _slightly_ better, given that `Node.DeepCopyAsNode()` guarantees that it returns a `Node`. 🤷🏻 
```go
defaultNode := node.DeepCopyAsNode()
typeCheck = t.AssignTypeTo(defaultNode)
if !typeCheck.HasViolations() {
	typeCheck = CheckNode(defaultNode)
}
defaultValue = defaultNode
```